goal_src/levels/swamp/billy.gc

```diff
@@ -5,3 +5,2048 @@
+        "launcher-97"
+        )
+       (0
+        kill
+        "swamp-rat-nest-5"
+        )
+       (0
+        kill
+        "swamp-rat-nest-4"
+        )
+       (0
+        kill
+        "swamp-rat-nest-3"
+        )
+       (0
+        kill
+        "swamp-rat-nest-2"
+        )
+       (0
+        kill
+        "sharkey-9"
+        )
+       (0
+        kill
+        "launcher-46"
+        )
+       (0
+        kill
+        "launcher-44"
+        )
+       (0
+        kill
+        "launcher-42"
+        )
+       (0
+        kill
+        "swamp-spike-43"
+        )
+       (0 kill "swamp-spike-44") (0 kill "swamp-spike-45")
+       )
+     )
+    )
+   (else
+    (if arg0
+     (format
+      0
+      "ERROR: <GMJ>: ~S playing anim for task status ~S~%"
+      (-> obj name)
+      (task-status->string (current-status (-> obj tasks)))
+      )
+     )
+    (-> obj draw art-group data 3)
+    )
+   )
+  )
+
+(defmethod get-art-elem billy ((obj billy))
+  (case (current-status (-> obj tasks)) 
+   (((task-status invalid) (task-status need-resolution))
+     (-> obj draw art-group data 10)
+     )
+   (else
+    (-> obj draw art-group data 3)
+    )
+   )
+  )
+
+(defmethod dummy-38 billy ((obj billy))
+  (case (current-status (-> obj tasks)) 
+   (((task-status need-reminder-a) (task-status need-reminder))
+     (go (method-of-object obj query))
+     )
+   (((task-status need-reward-speech))
+    (go (method-of-object obj play-anim))
+    )
+   (else
+    ((the-as (function nav-enemy none) (find-parent-method billy 38))
+     (the-as nav-enemy obj)
+     )
+    )
+   )
+  (none)
+  )
+
+(defmethod get-accept-anim billy ((obj billy) (arg0 symbol))
+  (if arg0
+   (close-current! (-> obj tasks))
+   )
+  (new 'static 'spool-anim
+   :name "billy-accept"
+   :index 6
+   :parts 3
+   :command-list '()
+   )
+  )
+
+(defmethod get-reject-anim billy ((obj billy) (arg0 symbol))
+  (new 'static 'spool-anim
+   :name "billy-reject"
+   :index 7
+   :parts 3
+   :command-list '()
+   )
+  )
+
+(defstate billy-done (billy)
+  :event
+  (behavior ((arg0 process) (arg1 int) (arg2 symbol) (arg3 event-message-block))
+   (let ((v1-0 arg2))
+    (the-as object (when (= v1-0 'billy-rat-needs-destination)
+                    (let* ((gp-0 arg0)
+                           (v1-2
+                            (if
+                             (and
+                              (nonzero? gp-0)
+                              (type-type? (-> gp-0 type) billy-rat)
+                              )
+                             gp-0
+                             )
+                            )
+                           )
+                     (when v1-2
+                      (set!
+                       (-> (the-as billy-rat v1-2) dest-type)
+                       (the-as uint 1)
+                       )
+                      (get-random-point
+                       (-> self path-waypts)
+                       (the-as vector (&-> v1-2 stack 400))
```

Comment:
hm, this is weird, are we sure about this type?

Reply:
missed a cast